src/nimblepkg/developfile.nim

```diff
@@ -393,7 +393,7 @@ proc load(path: Path, dependentPkg: PackageInfo, options: Options,
     # If this is a package develop file, but not a free one, for each of the
     # package's develop mode dependencies load its develop file if it is not
     # already loaded and merge its data to the current develop file's data.
-    for path, pkg in result.pathToPkg.dup:
```

Comment:
Here removing `dup` is also incorrect because below in `result.mergeFollowedDevFileData` a new values are being added to the `result.pathToPkg` table.